src/mds/MDLog.cc

```diff
@@ -147,18 +147,21 @@ uint64_t MDLog::get_safe_pos() const
   return journaler->get_write_safe_pos(); 
 }
 
-void MDLog::_log_estimated_complete_time(utime_t start_time, uint64_t start_pos){
+std::pair<double, uint64_t> MDLog::get_replay_estimated_complete_time(){
   auto end_pos = journaler->get_write_pos();
```

Comment:
`end_pos` isn't going to change during replay, do just fetch it once and use it for the computation.